### DISCUSSION TITLE
Stanford CS110L: Safety in Systems Programming - CS自学指南

### DISCUSSION THREAD

**Discussion by giscus** [PKUFlyingPig/cs-self-learning#91] [Announcements]
# Stanford CS110L: Safety in Systems Programming - CS自学指南

CS自学指南

https://csdiy.wiki/%E7%BC%96%E7%A8%8B%E5%85%A5%E9%97%A8/CS110L/

**Comment by azusachino** [comment]
学习打卡

**Reply by DavidZyy** [reply]
互勉

**Reply by HenryCROSS** [reply]
互勉x2

**Reply by Derick317** [reply]
Week 3 作业 Part 1 multi_pipe_test 的 file descriptor 真的是 0, 1, 2, 4, 5 吗？我直接运行了这个程序并用 /proc/xxx/fd 查看了文件描述符是 0, 1, 103, 2, 38, 39, 4, 40, 41, 5.

**Reply by TianWZhang** [reply]
俺也遇到了这个情况，看来是这样的

**Comment by yangqiong** [comment]
课程网站（2022）：https://web.stanford.edu/class/cs110l/
课程视频（2020 中文字幕）：https://www.bilibili.com/video/BV1Ra411A7kN?p=1&spm_id_from=333.788.b_6d756c74695f70616765.1

**Reply by HenryCROSS** [reply]
2022的作业，好像被上锁了？没办法克隆下来

**Reply by BeautyyuYanli** [reply]
> 2022的作业，好像被上锁了？没办法克隆下来

是的 建议阅读 2020 版本或混合阅读

**Reply by dengyunsheng250** [reply]
2022 winter好像开放了

**Comment by BeautyyuYanli** [comment]
这门课程的两个 proj 和 The Rust Programming Language 也是相同的

**Comment by Falicitas** [comment]
Week3 的远端服务器myth怎么登？
好奇做了 Week3 功课的同学是怎么完成的= =

**Reply by imabackstabber** [reply]
myth是给Stanford的学生使用的。
自己选择一台linux服务器可以解决环境的问题，实在不行可以使用wsl，不过具体的实现上会有针对wsl的compromise

**Comment by imabackstabber** [comment]
这门课的前置（或许不是）课程CS110是没有办法在校外登陆进去吗🤔

**Reply by imabackstabber** [reply]
看起来它的作业并不对外开放🗿

**Comment by King-ty** [comment]
请问有朋友的proj1出现行号不显示的问题吗？环境是WSL2 Ubuntu22.04

**Reply by King-ty** [reply]
已找到原因，是ubuntu22.04的问题，换成18以及20都可以正常运行

**Reply by GuoGuo614** [reply]
应该是gcc版本过新导致的，ArchLinux也遇到了这个问题，在Makefile里加上 `-gdwarf-4` 编译选项，也可以解决。

**Comment by Derick317** [comment]
Week 3 作业 Part 1 multi_pipe_test 的 file descriptor 真的是 0, 1, 2, 4, 5 吗？我直接运行了这个程序并用 /proc/xxx/fd 查看了文件描述符是 0, 1, 103, 2, 38, 39, 4, 40, 41, 5.

**Reply by iteng007** [reply]
也许你是在vscode的终端或者别的ide下的终端中执行的，试试直接在终端中执行，我一开始使用vscode的终端运行的结果也不是0,1,2,3,4,5。然后在Windows Terminal中执行就是test中要求的结果。

**Reply by TephrocactusMYC** [reply]
vscode会有.vscode-server的进程存在，所以会一直过不去。直接在虚拟机里运行测试即可

**Comment by new-ccl** [comment]
这个proj2不能正常编译了吗，还是我的环境有问题😭😭

**Comment by JEB-Bem** [comment]
project-1 中提到的 stack frames 相关的文章由于作者网站迁移丢失了，这是他的课程上最新的关于 stack frames 的 [notes](https://pk.org/classes/419/notes/memory-exploits.html) 以及[课程主页](https://pk.org/classes/419/notes/memory-exploits.html)
